agent/src/agent.ts

```
@@ -1320,6 +1325,15 @@ export class Agent extends MessageHandler implements ExtensionClient {
         }
     }
 
+    private newGlobalState(clientInfo: ClientInfo): AgentGlobalState {
+        if (clientInfo.capabilities?.globalState === 'server-managed') {
+            return new AgentGlobalState(
+                clientInfo.globalStatePath ?? join(codyPaths().data, 'globalState.json')
+            )
+        }
+        return new AgentGlobalState()
```

Let's throw an error if it's `client-managed`, it's OK to keep it unimplemented